packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -196,21 +236,26 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     // When the dropdown opens
     if (isOpen) {
       inputRef.current?.focus(); // Focus the input
-      setFilteredOptions(optionsRef.current); // Reset the filtered list
-      dispatch(['focus']);
+      // Reset the filtered list (does not reset when multiple enabled)
+      if (!isMultipleEnabled(selected, multiple)) {
+        setFilteredOptions(optionsRef.current);
+        dispatch({ type: 'focus' });
+      }
     }
     // When the dropdown closes
     else {
       // Reset the focused index
-      dispatch(['focus']);
-      // Reset the input value
-      setInputValue(
-        selectedIndex != undefined && selectedIndex >= 0
-          ? optionsRef.current[selectedIndex]?.label
-          : '',
-      );
+      dispatch({ type: 'focus' });
+      // Reset the input value if not multiple
+      if (!isMultipleEnabled(selected, multiple)) {
```

Comment:
Similarly, this can use `Array.isArray`

```suggestion
if (!multiple && !Array.isArray(selected)) {
```

Reply:
Added